recipes/qt/6.x.x/conanfile.py

```diff
@@ -659,14 +666,54 @@ def _create_plugin(pluginname, libname, type, requires):
-                self.cpp_info.components["qtCore"].frameworks.append("Cocoa")     # qtcore requires "_OBJC_CLASS_$_NSApplication" and more, which are in "Cocoa" framework
-                self.cpp_info.components["qtCore"].frameworks.append("Security")  # qtcore requires "_SecRequirementCreateWithString" and more, which are in "Security" framework
+                self.cpp_info.components["Qt6Core"].frameworks.append("IOKit")     # qtcore requires "_IORegistryEntryCreateCFProperty", "_IOServiceGetMatchingService" and much more which are in "IOKit" framework
+                self.cpp_info.components["Qt6Core"].frameworks.append("Cocoa")     # qtcore requires "_OBJC_CLASS_$_NSApplication" and more, which are in "Cocoa" framework
+                self.cpp_info.components["Qt6Core"].frameworks.append("Security")  # qtcore requires "_SecRequirementCreateWithString" and more, which are in "Security" framework
+
+        extension = ""
+        if self.settings.os == "Windows":
+            extension = ".exe"
+        filecontents = "set(QT_CMAKE_EXPORT_NAMESPACE Qt6)\n"
+        filecontents += "set(QT_VERSION_MAJOR %d)\n" % int(self.version.split(".")[0])
+        filecontents += "set(QT_VERSION_MINOR %d)\n" % int(self.version.split(".")[1])
+        filecontents += "set(QT_VERSION_PATCH %d)\n" % int(self.version.split(".")[2])
+        targets = ["moc", "rcc", "tracegen", "cmake_automoc_parser", "qlalr", "qmake"]
+        targets.extend(["qdbuscpp2xml", "qdbusxml2cpp"])
+        if self.options.gui:
+            targets.append("qvkgen")
+        if self.options.widgets:
+            targets.append("uic")
+        if self.options.qttools:
+            targets.extend(["qhelpgenerator", "qtattributionsscanner", "windeployqt"])
+            targets.extend(["lconvert", "lprodump", "lrelease", "lrelease-pro", "lupdate", "lupdate-pro"])
+        if self.options.qtshadertools:
+            targets.append("qsb")
+        if self.options.qtdeclarative:
+            targets.extend(["qmltyperegistrar", "qmlcachegen", "qmllint", "qmlimportscanner"])
+            targets.extend(["qmlformat", "qml", "qmlprofiler", "qmlpreview", "qmltestrunner"])
+        for target in targets:
+            filecontents += """\
+if(NOT TARGET ${{QT_CMAKE_EXPORT_NAMESPACE}}::{1})
+    add_executable(${{QT_CMAKE_EXPORT_NAMESPACE}}::{1} IMPORTED)
+    set_target_properties(${{QT_CMAKE_EXPORT_NAMESPACE}}::{1} PROPERTIES IMPORTED_LOCATION ${{CMAKE_CURRENT_LIST_DIR}}/../../../bin/{1}{0})
+endif()
+""".format(extension, target)
+        tools.save(os.path.join("lib", "cmake", "Qt6Core", "extras.cmake"), filecontents)
+
+        self.cpp_info.components["Qt6Core"].builddirs.append(os.path.join("res","archdatadir","bin"))
+        for m in os.listdir(os.path.join("lib", "cmake")):
+            module = os.path.join("lib", "cmake", m, "%sMacros.cmake" % m)
+            if os.path.isfile(module):
+                self.cpp_info.components[m].build_modules.append(module)
```

Comment:
`build_modules` support generator specific configurations.... 

I think you might want to checkout https://github.com/conan-io/conan-center-index/pull/4795 which was similar to this PR and done by our expert

...which you will be added to the list once this is merged.

Reply:
Haha not sure about the expert status, copied most of this from @ericLemanissier. As for the generator specific configurations. 

What would the advantage be of adding it to certain generators only? Because the modules are cmake only?

Reply:
Exactly, it would be very weird to have cmake files include for the other generators

Reply:
Ok so the cmake, cmake find package and cmake find package multi generators correct?

Reply:
just find and fin_multi

Reply:
Fixed